cli/args/flags.rs

```diff
@@ -2206,23 +2215,31 @@ diagnostic errors from remote modules will be ignored.",
     )
 }
 
-fn check_arg() -> Arg {
-  Arg::new("check")
+fn check_arg(checks_local_by_default: bool) -> Arg {
+  let arg = Arg::new("check")
     .conflicts_with("no-check")
     .long("check")
     .num_args(0..=1)
     .require_equals(true)
     .value_name("CHECK_TYPE")
-    .help("Type-check modules")
-    .long_help(
-      "Type-check modules.
-
-Deno does not type-check modules automatically from v1.23 onwards. Pass this
-flag to enable type-checking or use the 'deno check' subcommand.
+    .help("Type-check modules");
 
+  if checks_local_by_default {
+    arg.long_help(
+        "Enable type-checking. This subcommand type-checks local modules by default.
```

Comment:
`Enable type-checking.`
Seems a bit confusing given that its already enabled

Reply:
How about `Set type-checking behavior` instead? And should it mention that using `--check` with these subcommands is redundant since it is enabled by default?

Reply:
yes, that sounds good